nano/secure/ledger.cpp

```diff
@@ -675,6 +684,18 @@ check_bootstrap_weights (true),
 epoch_link (epoch_link_a),
 epoch_signer (epoch_signer_a)
 {
+	if (!store.init_error ())
+	{
+		auto transaction = store.tx_begin_read ();
+		for (auto i (store.latest_begin (transaction)), n (store.latest_end ()); i != n; ++i)
+		{
+			nano::account_info const & info (i->second);
+			nano::block_hash rep_block (representative_calculated (transaction, info.head));
+			auto source_block (store.block_get (transaction, rep_block));
+			auto source_rep (source_block->representative ());
+			rep_weights.representation_add (source_rep, info.balance.number ());
```

Comment:
I think here we can go with
```cpp
auto block (store.block_get (transaction, info.rep_block));
assert (block != nullptr);
rep_weights.representation_add (block->account (), info.balance.number ());
```

It should be faster than the visitor used by `representative_calculated`

Reply:
Yes good point. I had copied that from `--debug_dump_representatives`, `info.rep_block` has already cached it so can be used.